CMakeLists.txt

```diff
@@ -909,6 +909,7 @@ check_include_file_concat("net/if.h"         HAVE_NET_IF_H)
 check_include_file_concat("netdb.h"          HAVE_NETDB_H)
 check_include_file_concat("netinet/in.h"     HAVE_NETINET_IN_H)
 check_include_file_concat("netinet/tcp.h"    HAVE_NETINET_TCP_H)
+check_include_file("linux/tcp.h"      HAVE_LINUX_TCP_H)
```

Comment:
This is duplicating changes from #6262 ...

Reply:
that is true. this PR does not work without the other changes. merging the identical changes should not be a problem.

Reply:
Merging yes, but adding irrelevant changes into a PR makes it bad and not possible to review/ack correctly as we don't know what's the PR and what's not the PR.

Reply:
Since #6262 is merged now, you can rebase this and force-push to clean those up.